cpp/src/parquet/metadata.h

```diff
@@ -452,8 +453,12 @@ class PARQUET_EXPORT ColumnChunkMetaDataBuilder {
   // column chunk
   // Used when a dataset is spread across multiple files
   void set_file_path(const std::string& path);
+
   // column metadata
   void SetStatistics(const EncodedStatistics& stats);
+
+  KeyValueMetadata& key_value_metadata();
```

Comment:
Please let's avoid non-const references, as per our C++ conventions. Instead, you could follow the current builder pattern:
```suggestion
  void SetKeyValueMetadata(const std::shared_ptr<const KeyValueMetadata>&);
```